All test_copy_sendfile tests fail with Python 3.9

Hi,
It looks like some tests from the git master fail with Python 3.9.0rc1 (they are fine with 3.8.5).

```
============================================ test session starts =============================================
platform linux -- Python 3.9.0rc1, pytest-6.0.1, py-1.9.0, pluggy-0.13.1 -- /usr/bin/python3.9
cachedir: .pytest_cache
rootdir: /tmp/pyfilesystem2, configfile: setup.cfg
plugins: requests-mock-1.8.0
collecting ... collected 1777 items
================================================== FAILURES ==================================================
________________________________________ TestOSFS.test_copy_sendfile _________________________________________

self = <tests.test_osfs.TestOSFS testMethod=test_copy_sendfile>

    @pytest.mark.skipif(osfs.sendfile is None, reason="sendfile not supported")
    def test_copy_sendfile(self):
        # try copying using sendfile
        with mock.patch.object(osfs, "sendfile") as sendfile:
            sendfile.side_effect = OSError(errno.ENOSYS, "sendfile not supported")
            self.test_copy()
        # check other errors are transmitted
        self.fs.touch("foo")
        with mock.patch.object(osfs, "sendfile") as sendfile:
            sendfile.side_effect = OSError(errno.EWOULDBLOCK)
            with self.assertRaises(OSError):
>               self.fs.copy("foo", "foo_copy")
E               AssertionError: OSError not raised

tests/test_osfs.py:101: AssertionError
________________________________________ TestOSFS.test_copy_sendfile _________________________________________

self = <tests.test_osfs.TestOSFS testMethod=test_copy_sendfile>

    @pytest.mark.skipif(osfs.sendfile is None, reason="sendfile not supported")
    def test_copy_sendfile(self):
        # try copying using sendfile
        with mock.patch.object(osfs, "sendfile") as sendfile:
            sendfile.side_effect = OSError(errno.ENOSYS, "sendfile not supported")
            self.test_copy()
        # check other errors are transmitted
        self.fs.touch("foo")
        with mock.patch.object(osfs, "sendfile") as sendfile:
            sendfile.side_effect = OSError(errno.EWOULDBLOCK)
            with self.assertRaises(OSError):
>               self.fs.copy("foo", "foo_copy")
E               AssertionError: OSError not raised

tests/test_osfs.py:101: AssertionError
________________________________________ TestSubFS.test_copy_sendfile ________________________________________

self = <tests.test_subfs.TestSubFS testMethod=test_copy_sendfile>

    @pytest.mark.skipif(osfs.sendfile is None, reason="sendfile not supported")
    def test_copy_sendfile(self):
        # try copying using sendfile
        with mock.patch.object(osfs, "sendfile") as sendfile:
            sendfile.side_effect = OSError(errno.ENOSYS, "sendfile not supported")
            self.test_copy()
        # check other errors are transmitted
        self.fs.touch("foo")
        with mock.patch.object(osfs, "sendfile") as sendfile:
            sendfile.side_effect = OSError(errno.EWOULDBLOCK)
            with self.assertRaises(OSError):
>               self.fs.copy("foo", "foo_copy")
E               AssertionError: OSError not raised

tests/test_osfs.py:101: AssertionError
________________________________________ TestOSFS.test_copy_sendfile _________________________________________

self = <tests.test_osfs.TestOSFS testMethod=test_copy_sendfile>

    @pytest.mark.skipif(osfs.sendfile is None, reason="sendfile not supported")
    def test_copy_sendfile(self):
        # try copying using sendfile
        with mock.patch.object(osfs, "sendfile") as sendfile:
            sendfile.side_effect = OSError(errno.ENOSYS, "sendfile not supported")
            self.test_copy()
        # check other errors are transmitted
        self.fs.touch("foo")
        with mock.patch.object(osfs, "sendfile") as sendfile:
            sendfile.side_effect = OSError(errno.EWOULDBLOCK)
            with self.assertRaises(OSError):
>               self.fs.copy("foo", "foo_copy")
E               AssertionError: OSError not raised

tests/test_osfs.py:101: AssertionError
_______________________________________ TestTempFS.test_copy_sendfile ________________________________________

self = <tests.test_tempfs.TestTempFS testMethod=test_copy_sendfile>

    @pytest.mark.skipif(osfs.sendfile is None, reason="sendfile not supported")
    def test_copy_sendfile(self):
        # try copying using sendfile
        with mock.patch.object(osfs, "sendfile") as sendfile:
            sendfile.side_effect = OSError(errno.ENOSYS, "sendfile not supported")
            self.test_copy()
        # check other errors are transmitted
        self.fs.touch("foo")
        with mock.patch.object(osfs, "sendfile") as sendfile:
            sendfile.side_effect = OSError(errno.EWOULDBLOCK)
            with self.assertRaises(OSError):
>               self.fs.copy("foo", "foo_copy")
E               AssertionError: OSError not raised

tests/test_osfs.py:101: AssertionError

========================================== short test summary info ===========================================
FAILED tests/test_osfs.py::TestOSFS::test_copy_sendfile - AssertionError: OSError not raised
FAILED tests/test_subfs.py::TestOSFS::test_copy_sendfile - AssertionError: OSError not raised
FAILED tests/test_subfs.py::TestSubFS::test_copy_sendfile - AssertionError: OSError not raised
FAILED tests/test_tempfs.py::TestOSFS::test_copy_sendfile - AssertionError: OSError not raised
FAILED tests/test_tempfs.py::TestTempFS::test_copy_sendfile - AssertionError: OSError not raised
=========================== 5 failed, 1756 passed, 16 skipped, 1 warning in 18.94s ===========================
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Start testing on PyPy. Due to [#342](https://github.com/PyFilesystem/pyfilesystem2/issues/342)
   we have to treat PyPy builds specially and allow them to fail, but at least we'll
   be able to see if we break something aside from known issues with FTP tests.
+- Stop patching copy with Python 3.8+ because it already uses sendfile.
 
 ## [2.4.11] - 2019-09-07
 

--- a/CONTRIBUTORS.md
+++ b/CONTRIBUTORS.md
@@ -6,6 +6,7 @@ Many thanks to the following developers for contributing to this project:
 - [Diego Argueta](https://github.com/dargueta)
 - [Geoff Jukes](https://github.com/geoffjukes)
 - [Giampaolo](https://github.com/gpcimino)
+- [Louis Sautier](https://github.com/sbraz)
 - [Martin Larralde](https://github.com/althonos)
 - [Will McGugan](https://github.com/willmcgugan)
 - [Zmej Serow](https://github.com/zmej-serow)

--- a/fs/osfs.py
+++ b/fs/osfs.py
@@ -419,7 +419,7 @@ class OSFS(FS):
             raise errors.DirectoryExpected(dirname(dst_path))
         return _src_path, _dst_path
 
-    if sys.version_info[:2] <= (3, 8) and sendfile is not None:
+    if sys.version_info[:2] < (3, 8) and sendfile is not None:
 
         _sendfile_error_codes = {
             errno.EIO,

--- a/tests/test_osfs.py
+++ b/tests/test_osfs.py
@@ -6,6 +6,7 @@ import io
 import os
 import shutil
 import tempfile
+import sys
 import unittest
 import pytest
 
@@ -88,6 +89,11 @@ class TestOSFS(FSTestCases, unittest.TestCase):
         self.assertNotIn("TYRIONLANISTER", fs2.getsyspath("/"))
 
     @pytest.mark.skipif(osfs.sendfile is None, reason="sendfile not supported")
+    @pytest.mark.skipif(
+        sys.version_info >= (3, 8),
+        reason="the copy function uses sendfile in Python 3.8+, "
+        "making the patched implementation irrelevant",
+    )
     def test_copy_sendfile(self):
         # try copying using sendfile
         with mock.patch.object(osfs, "sendfile") as sendfile:

